### ISSUE TITLE
Upgrade rollup to v2

### ISSUE THREAD

**Issue by shirotech** [AlexxNB/svelte-docs#24]
Seems we are still in v1, and it's quite hard adding my own plugins and configurations to the build. Having said that it requires updating some of the existing plugins that you have as there are some breaking changes. For that, I'll really need your help on that one, hope you're happy to do it. Many thanks 😄

**Comment by AlexxNB** [comment]
Rollup using by `@svelte-docs/core` only. Users can't use their own `rollup.config.js`. What existing plugins you want to use and how? 
To move on Rollup2 we should rewrite huge amount builtin plugins in `core` but as I say in https://github.com/AlexxNB/svelte-docs/issues/20, I'm going to rewrite `core` and rewriting plugins in this package may bu useless thing right now

**Comment by shirotech** [comment]
Example this is my `rollup.config.js`

```js
import {htmlResolver, resolvePlugins} from '...';
import conf from '@svelte-docs/core/rollup.config';

const extraPlugins = resolvePlugins(['resolver', 'conf'], {resolver: htmlResolver});
conf[0].plugins.unshift(...extraPlugins); // injecting plugins to main
conf[1].plugins.unshift(...extraPlugins); // injecting plugins to example

conf.push(); // this part here breaks, as v1 doesn't support code splitting using object in input
// https://rollupjs.org/guide/en/#input
// 

export default conf;
```

e.g. v1 doesn't support this kind of configuration

```js
// rollup.config.js
export default {
  ...,
  input: {
    a: 'src/main-a.js',
    'b/index': 'src/main-b.js'
  },
  output: {
    ...,
    entryFileNames: 'entry-[name].js'
  }
};
```

> To move on Rollup2 we should rewrite huge amount builtin plugins in `core` but as I say in #20, I'm going to rewrite `core` and rewriting plugins in this package may bu useless thing right now

If that's the case, I'll look for an alternative solution.

**Comment by AlexxNB** [comment]
It is tricky and undocumented =)
You may check `core` plugins, if there are not so much changes and you have a time for test this, feel free to PR.

**Comment by shirotech** [comment]
I feel bad hacking the project, but really need it to build using the builtin `example` which imports directly from a component, but that said component requires additional steps etc.

Cool, I'll see if I can upgrade it cleanly, last time I did it wasn't too bad. Will PR latest this week as for a heads up. Cheers.

**Comment by AlexxNB** [comment]
Yeah! I'm waiting for release to see if it is possible to recreate svelte-docs with svelte-kit.
